services/client/src/pages/events/reducers/events.js

```diff
@@ -23,14 +24,22 @@ export function events(state = defaultState, action) {
     case "EVENTS_FETCH_DATA_SUCCESS": {
       const { data } = action.events;
 
-      const upcomingEvents = updateEventsList(data.upcoming_events);
-      const recentEvents = updateEventsList(data.recent_events);
+      const upcomingEvents = updateEventsList([
+        ...state.upcomingEvents,
+        ...data.upcoming_events
+      ]);
+      const recentEvents = updateEventsList([
+        ...state.recentEvents,
+        ...data.recent_events
+      ]);
 
       return {
         ...state,
         upcomingEvents,
         recentEvents,
-        isLoading: false
+        isLoading: false,
+        hasMoreItems:
```

Comment:
This should come from back-end to avoid the last call.

Reply:
Worth updating the backend for this PR? Would be nice if it provided some meta data on number of items, number of pages, etc.

Reply:
I'll raise an issue to update the backend to provide Meta data for paginated endpoints and a follow up to refractor.